hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/impl/ERCReadOnlyAbstractPrecompile.java

```diff
@@ -20,17 +20,22 @@
  * ‍
  */
 
+import com.hedera.services.context.primitives.StateView;
+import com.hedera.services.fees.FeeCalculator;
+import com.hedera.services.fees.calculation.UsagePricesProvider;
 import com.hedera.services.store.contracts.WorldLedgers;
 import com.hedera.services.store.contracts.precompile.Precompile;
 import com.hedera.services.store.contracts.precompile.SyntheticTxnFactory;
 import com.hedera.services.store.contracts.precompile.codec.DecodingFacade;
 import com.hedera.services.store.contracts.precompile.codec.EncodingFacade;
+import com.hedera.services.store.contracts.precompile.utils.PrecompilePricingUtils;
 import com.hederahashgraph.api.proto.java.Timestamp;
 import com.hederahashgraph.api.proto.java.TokenID;
 import com.hederahashgraph.api.proto.java.TransactionBody;
 import org.apache.tuweni.bytes.Bytes;
 import org.hyperledger.besu.evm.frame.MessageFrame;
 
+import javax.inject.Provider;
 import java.util.function.UnaryOperator;
 
 public abstract class ERCReadOnlyAbstractPrecompile implements Precompile {
```

Comment:
We shouldn't restrict precompiles that extend this abstract class to be only ERC ones.

We can rename it: `ERCReadOnlyAbstractPrecompile ` -> `AbstractReadOnlyPrecompile`